src/framebus.ts

```diff
@@ -29,18 +32,34 @@ const DefaultPromise = (typeof window !== "undefined" &&
 export class Framebus {
   origin: string;
   channel: string;
+  targetFrames: IFrameOrWindowList;
 
   private verifyDomain?: VerifyDomainMethod;
   private isDestroyed: boolean;
   private listeners: Listener[];
+  private hasAdditionalChecksForOnListeners: boolean;
+  private limitBroadcastToFramesArray: boolean;
 
   constructor(options: FramebusOptions = {}) {
     this.origin = options.origin || "*";
     this.channel = options.channel || "";
     this.verifyDomain = options.verifyDomain;
 
+    // if targetFrames is not used, we will be broadcasting
+    // to the top level window or to itself.
+    // By default, the broadcast function will loop through
+    // all the known siblings and children of the window.
+    // If a targetFrames array is passed, it will instead
+    // only broadcast to those targetFrames
+    this.targetFrames = options.targetFrames || [window.top || window.self];
```

Comment:
```suggestion
    // if a targetFrames configuration is not passed in,
    // the default behavior is to broadcast the payload
    // to the top level window or to the frame itself.
    // By default, the broadcast function will loop through
    // all the known siblings and children of the window.
    // If a targetFrames array is passed, it will instead
    // only broadcast to those specified targetFrames
    this.targetFrames = options.targetFrames || [];
```